Extensions/soft_refresh.js

```diff
@@ -11,212 +11,184 @@ XKit.extensions.soft_refresh = new Object({
-
+	clicked: function(event) {
+		if (event.altKey) {
+			location.reload(true);
+		} else {
+			XKit.extensions.soft_refresh.start_load();
+		}
 	},
 
-	do_post_ids: function() {
+	start_load: function() {
+		if (this.loading) { return; }
+		this.loading = true;
 
-		$(".posts .post").each(function() {
+		$("html, body").animate({ scrollTop: 0 }, "slow");
 
-			$(this).parent().attr('data-xkit-post-id', $(this).attr('data-post-id'));
+		$("#new_post").after('<div id="xkit_soft_refresh">Checking for new posts</div>');
+		$("#xkit_soft_refresh").slideDown("fast");
 
+		$("#new_post_notice_container").css({
+			transform: "scale(0)",
+			opacity: 0
 		});
 
+		this.top_post = $("#posts > li.post_container:not(#new_post_buttons)").first();
+		this.request("");
 	},
 
-	logo_clicked: function(event) {
-
-		if (event.altKey) {
-			location.reload(true);
-			return;
-		}
-		XKit.extensions.soft_refresh.load_posts();
-
-	},
-
-	load_posts: function() {
-
-		if (typeof XKit.extensions.filter_by_type !== "undefined") {
-			if (XKit.extensions.filter_by_type.running === true) {
-				if (XKit.extensions.filter_by_type.current_filter !== "0") {
-					location.reload(true);
+	request: function(page) {
+		fetch(`https://www.tumblr.com/dashboard${page}`)
+			.then(response => {
+				if (!response.ok) {
+					this.show_error();
 					return;
 				}
-			}
-		}
 
-		if (this.loading === true) { return; }
-		this.loading = true;
-
-		$("#new_post").after("<div id=\"xkit_soft_refresh\">Checking for new posts</div>");
-		$("#xkit_soft_refresh").slideDown('fast');
-
-		document.title = this.default_page_title;
-		$("#new_post_notice_container").css("display", "none");
-		$("#new_post_notice_container .tab_notice_value").html("0");
-
-		function soft_refresh_hit_triggers() {
-			var postIds = add_tag;
-			var Tumblr = window.Tumblr || window.top.Tumblr;
-
-			try {
-				postIds.forEach(function(id) {
-					var postId = "post_" + id;
-					var postElement = jQuery(document.getElementById(postId));
-					var fakeView = new Tumblr.PostView({
-						el: postElement,
-						model: Tumblr.PostsView.prototype.createPostModelFromEl(postElement)
-					});
-					Tumblr.Events.trigger("postsView:createPost", fakeView);
-				});
-			} catch (e) {
-				console.warn("soft_refresh new approach failed:", e);
-			}
-			Tumblr.Events.trigger("posts:load");
-			Tumblr.Events.trigger("DOMEventor:updateRect");
-			Tumblr.AudioPlayer.update_all();
-		}
+				let end = false;
+
+				response.text().then(responseText => {
+					if (page === "") {
+						$("#new_post_notice_container .tab_notice_value").html("0");
+						document.title = this.default_page_title;
+						$("#new_post_notice_container")
+							.removeClass("tab-notice--active")
+							.removeAttr("style");
+						// Remove every notification between the new post buttons and the first existing post
+						$("#posts > li.notification:not(.post_container:not(#new_post_buttons) ~ .notification)").remove();
```

Comment:
because notifications have no unique attributes like posts do, so we can't update blocks of notifications. but what we *can* do is remove the first block of notifications and then fetch *everything* above the first post on the dashboard, which would include the notifications we removed and additional notifications that now belong to that block.